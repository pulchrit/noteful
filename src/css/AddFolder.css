
.add-folder-section { 
    display: flex;
    flex-direction: column;
    width: 50%;
    margin: 20px auto;
    padding: 20px;
    border: 1px solid lightgrey;
    border-radius: 4px;

}

.add-folder-error {
    color: red;
}

.add-form-title {
    font-size: 1.5em;
    text-align: center;
    color: black;
}

.add-folder-form {
    background-color: lightgrey;
    border-radius: 4px;
    padding: 20px;
    width: 98%;
    margin: 0 auto;
    display: flex;
    flex-direction: column;
    align-items: center;
  
}

.folder-name-label, .folder-name-input {
    display: block;
    width: 95%;
    margin: 10px 0;
    line-height: 2em;
    font-size: 1em;
}

.folder-name-input {
    padding: 5px;
}

.add-folder-submit-button:disabled {
    background-color: grey;
}

.add-folder-submit-button {
    padding: 15px;
    width: 80px;
    background-color: green;
    color: white;
    border-radius: 4px;
    font-weight: bold;
    font-size: 1em;
}




