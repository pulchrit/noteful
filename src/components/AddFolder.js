import React from 'react';
import "../css/AddFolder.css";

const AddFolder = () => (
    <h2 className='add-form-title'>Add folder here</h2>
)

export default AddFolder;