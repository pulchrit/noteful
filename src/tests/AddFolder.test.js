import React from 'react';
import ReactDOM from 'react-dom';
import {BrowserRouter} from 'react-router-dom';
import toJson from 'enzyme-to-json';
import {shallow} from 'enzyme';
import AddFolder from '../components/AddFolder';

/* describe("AddFolder component", () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <BrowserRouter>
        <AddFolder />
      </BrowserRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders UI as expected', () => {
    const wrapper = shallow(<AddFolder />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
}); */
