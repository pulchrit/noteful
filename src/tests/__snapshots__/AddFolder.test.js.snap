// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddFolder component renders UI as expected 1`] = `
<section
  className="add-folder-section"
>
  <h2
    className="add-form-title"
  >
    Add a folder
  </h2>
  <div
    className="add-folder-error"
    role="alert"
  />
  <form
    className="add-folder-form"
    onSubmit={[Function]}
  >
    <label
      className="folder-name-label"
      htmlFor="folderName"
    >
      Please enter a name for your folder:
      <input
        aria-describedby="folder-name-error"
        aria-label="Name for the new folder"
        aria-required="true"
        className="folder-name-input"
        id="folderName"
        name="folderName"
        onChange={[Function]}
        type="text"
      />
      <ValidationError
        errorName="folder-name-error"
        notValid={true}
        validationMessage=""
      />
    </label>
    <button
      className="add-folder-submit-button"
      disabled={true}
      type="submit"
    >
      Save
    </button>
  </form>
</section>
`;
